### PR TITLE
Mod2 2

### PR DESCRIPTION
**Pull request by ryankbales**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -9,5 +9,14 @@ def current_user
   	User.find(session[:user_id]) if session[:user_id]
   end
 
-  helper_method :current_user
+  def average_rating(obj)
```
Comment (matthew-ta) [inline app/controllers/application_controller.rb]: Maybe another word instead of object here.

File: app/controllers/queue_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if current_user.queue_items.include?(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def queue_video(video)
+    QueueItem.create(video: video, user: current_user, position: new_queue_item_position) unless current_user_queued_video?(video)
```
Comment (matthew-ta) [inline app/controllers/queue_items_controller.rb]: This line http://cl.ly/image/3r2X0i0d4620 is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits

File: app/controllers/queue_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def new_queue_item_position
+    current_user.queue_items.count + 1
+  end
+
+  def current_user_queued_video?(video)
+    current_user.queue_items.map(&:video).include?(video)
```
Comment (matthew-ta) [inline app/controllers/queue_items_controller.rb]: I would move this to the user model - but that is just a personal thing.

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if @review.save
+      flash[:notice] = "Your review has been posted."
+    else
+      flash[:errors] = "You need to enter both a rating and review."
+    end
+    redirect_to video_path(@video)
+  end
+
+  def edit
+    @review = Review.find(params[:id])
```
Comment (matthew-ta) [inline app/controllers/reviews_controller.rb]: Should the user only be able to edit their own review? If so this should be something like `@review = current_user.reviews.where(id: params[:id]).first`

File: app/controllers/reviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @video = @review.video
+  end
+
+  def update
+    @review = Review.find(params[:id])
```
Comment (matthew-ta) [inline app/controllers/reviews_controller.rb]: Same as above here.

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -12,6 +12,10 @@ def create
 		end
 	end
 
+	def show
+		@user = current_user
```
Comment (matthew-ta) [inline app/controllers/users_controller.rb]: There is an indentation issue here

File: app/controllers/videos_controller.rb [hunk 7]
```diff
@@ -13,6 +13,7 @@ def create
 
 	def show
 		@video = Video.find(params[:id])
+		@review = Review.new
```
Comment (matthew-ta) [inline app/controllers/videos_controller.rb]: There is an indentation issue here

File: app/models/category.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 class Category < ActiveRecord::Base
 	has_many :videos, -> { order("created_at DESC")}
+  validates_presence_of :name
```
Comment (matthew-ta) [inline app/models/category.rb]: Indentation issues in this file.

File: app/models/user.rb [hunk 9]
```diff
@@ -3,4 +3,7 @@ class User < ActiveRecord::Base
 	validates_uniqueness_of :email
 
 	has_secure_password
```
Comment (matthew-ta) [inline app/models/user.rb]: Indentation issue

File: app/models/video.rb [hunk 10]
```diff
@@ -1,6 +1,6 @@
 class Video < ActiveRecord::Base
 	belongs_to :category
-
+	has_many :reviews
 	validates_presence_of :title, :description
```
Comment (matthew-ta) [inline app/models/video.rb]: Indentation issue in this file.

File: spec/controllers/reviews_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+      it "sets the flash notice" do
+        expect(flash[:notice]).not_to be_blank
+      end
+    end
+  end
+
+  context "when invalid input" do
+    let(:video) { Fabricate(:video) }
+    before do
+      post :create, video_id: video.id, review: { review: "This is a post witout a rating.", video_id: video.id , user_id: Fabricate(:user).id }
```
Comment (matthew-ta) [inline spec/controllers/reviews_controller_spec.rb]: You can move the hash here into a local variable to reduce line length

File: spec/spec_helper.rb [hunk 12]
```diff
@@ -58,4 +58,5 @@
   # The different available types are documented in the features, such as in
   # https://relishapp.com/rspec/rspec-rails/v/3-0/docs
   config.infer_spec_type_from_file_location!
+  # config.render_views = true
```
Comment (matthew-ta) [inline spec/spec_helper.rb]: I suggest removing this line if it isn't in use.

**Comment by ryankbales** [comment]
@matthew-ta Refactored and cleaned up.  Thanks!

**Pull request merged**